src/test/lambda/wizards/stateController.test.ts

```diff
@@ -0,0 +1,217 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as sinon from 'sinon'
```

Comment:
A note on `sinon`:
The tests here are _not_ manipulating the unit under test using stubs. Rather, stubs are being used as a convenient way to create functions and make assertions on how the functions were used. Testing could instead be done with 'custom' functions which I might revisit, though I don't think it's really necessary to express functionality. Stubs already do the same and more of whatever function I could create.

Reply:
> Testing could instead be done with 'custom' functions which I might revisit, though I don't think it's really necessary to express functionality of the controller. Stubs already do the same and more of whatever function I could create.

I like the idea of avoiding sinon entirely because it avoids others from unconsciously mocking things in these tests. Fewer tests doing `import sinon` helps to see at a glance where mocks are being (not) used.